Doc/c-api/typeobj.rst

```diff
@@ -2148,21 +2148,20 @@ and :c:data:`PyType_Type` effectively act as defaults.)
    to be called before attempting to break reference cycles, ensuring
    that it finds the object in a sane state.
 
-   :c:member:`~PyTypeObject.tp_finalize` should not mutate the current exception status;
-   therefore, a recommended way to write a non-trivial finalizer is::
+   :c:member:`~PyTypeObject.tp_finalize` should not mutate the current exception
+   status. Non-trivial finalizers should use :c:func:`PyErr_GetRaisedException`,
```

Comment:
"non-trivial" is not accurate. You can have a lot of code which doesn't raise exceptions, and short code which raises exceptions. I prefer to keep the previous sentence.